### ISSUE TITLE
Extra packs for Orcsoberfest Island vs. Percht Island

### ISSUE THREAD

**Issue by mathiasbynens** [tibiamaps/tibia-map-data#104]
Our map data could default to showing Orcsoberfest Island all year through (since Orcsoberfest happens twice per year), except in December and January (since Winterlight Solstice takes place from December 22 to January 10 each year).

We could set up automation to update our map data periodically.

**Comment by mathiasbynens** [comment]
Given that:

- Orcsoberfest Island is accessible from the second Friday of October to the second Friday of March.
- Percht Island is accessible from December 22 to January 10 each year.

I'd suggest something like the following:

- On December 1st, we enable the Percht Island pack.
- On February 1st, we enable the Orcsoberfest Island pack.

@Hunter-of-Dragoes @LeeKunTibia Let me know if you'd prefer other dates (for example, to keep Percht Island visible for slightly longer).

**Comment by mathiasbynens** [comment]
Test commits, triggered by the new GitHub Actions:

- [Enable Orcsoberfest Island maps](https://github.com/tibiamaps/tibia-map-data/commit/0761a5ae4a8b7e0a82caf50d57e5be6a3e08ad2d)
- [Enable Percht Island maps](https://github.com/tibiamaps/tibia-map-data/commit/fd8b6177028f0881dcd1bc30ddcecf86e1b7f4a3)

Seems to work great!

**Comment by Hunter-of-Dragoes** [comment]
Actually, I think we need a more robust solution.

Tibia map has become dynamic more than ever. Differently from temporary map changes caused by some mini world changes and which basically consisted of adding a portal, a bridge, some stairs or other objects, now we have map areas that are completely different depending on the time of the year. In my point of view, this seasonal change should be approached on the online Mapper with the implementation of a time dimension in addition to the existing three physical dimensions (x, y and z). This means that each map layer could have, if necessary, an alternative version.

This way, we could have different versions of other places, such as the Halls of Hope at north of Thais (implemented with Forgotten Knowledge Quest) and areas affected by MWCs and events, not just the Orcsoberfest/Winterlight Solstice island.

Since we're looking into this, we could also have a version of the map with some old places which got removed from the game, this would be especially useful to point the location of removed NPCs such Elgar at the old Travora island and The Gatekeeper on Rookgaard's premium area.

Unnecessary to say that the swapping between "time dimensions" should be user-friendly both on TibiaMaps and TibiaWiki. Even though this change would make it require more time to manage the static versions of the map - which is the great majority of it - I do see good points in implementing it.

Regarding the downloadable map files, I think players should be given the options to download the Orcsoberfest version or Winterlight Solstice version. Additionally, as Mathias said, depending on the time of the year, one of them could be the default downloadable version, e.g the Winterlight Solstice version could be the default from October 21st till January 10th.

**Comment by mathiasbynens** [comment]
I agree, that would be a nice solution. It's also a lot of work :) Orcsoberfest Island / Percht Island seemed like the more pressing issue compared to MWCs, so that's what we focused on first.

Overall, this "extra packs" functionality seems like a good fit for what you're suggesting, except that it doesn't directly express a time dimension, but can also be used for other things (e.g. marker-only packs, etc.). 

#49 is an example of something that would make sense as an optional pack. The relevant markers are already in [the `achievements` pack](https://github.com/tibiamaps/tibia-map-data/tree/main/extra/achievements), but if we want to introduce a "mini world changes" pack, or even a separate `mwc-thawing` pack, we could do that.

We should probably add support for transparency in pack PNGs, so that we can overwrite only those pixels that need overwriting.

**Comment by LeeKunTibia** [comment]
Switching from Winterlight Solstice to Orcsoberfest is deleting areas of the Soul War Quest (Ebb and Flow and Claustrophobic Inferno):

https://github.com/tibiamaps/tibia-map-data/commit/7af7b01ed7f081b311a5dde0438e7506ca52ae89#diff-e2a5c803b87b5d32b9a3cc42fd0be23936b735157cac50c0663f70500d42b00d

**Comment by mathiasbynens** [comment]
@LeeKunTibia Thanks for catching that! Fixed.

**Comment by LeeKunTibia** [comment]
Found another issue caused by this: Tentugly's boat is removed when the pack is switched: https://github.com/tibiamaps/tibia-map-data/commit/85391ae2ecee691ffd63088fc7a875b14507f359?short_path=a852e1d#diff-a852e1d457ade61101e2765b129546293153474758e997260419bf0212228e30

https://tibiamaps.io/map#33726,31189,7:1

It also affects the floors above ground because of the mast (explored using the Cyclopedia Map).

**Comment by mathiasbynens** [comment]
Version with the boat: https://raw.githubusercontent.com/tibiamaps/tibia-map-data/72b69f2024675533bad4f0ec6a240b438841fa7e/data/floor-07-map.png (and similar for other floors)

**Comment by mathiasbynens** [comment]
> It also affects the floors above ground because of the mast (explored using the Cyclopedia Map).

Hmm, it looks like we actually didn't have the mast explored in that version?

https://raw.githubusercontent.com/tibiamaps/tibia-map-data/72b69f2024675533bad4f0ec6a240b438841fa7e/data/floor-06-map.png

**Comment by mathiasbynens** [comment]
944665d70f441d003b3a357c2745c767291bc417 restores the ground floor, at least. We still need the upper floor + the mast, both of which seem to be missing @ 72b69f2024675533bad4f0ec6a240b438841fa7e.

**Comment by LeeKunTibia** [comment]
Here are the upper floor + mast.

Important detail: the mast is from the Cyclopedia, so ti doesn't have the temporary areas in them. This may require some manual tweaking before you can properly add the maps to the packs. 

[Tentugly.zip](https://github.com/tibiamaps/tibia-map-data/files/9042994/Tentugly.zip)
